### ISSUE TITLE
☂️ Linter: eslint-plugin-react-perf

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#2041]
https://github.com/cvazac/eslint-plugin-react-perf

## Getting Started

```sh
just new-react-perf-rule <RULE NAME>
```

This will  create a `<rule name>.rs` file in `crates/oxc_linter/src/rules/react-perf`. Then, register the rule in `crates/oxc_linter/src/rules.rs` by adding the name to the `react-perf` module and in `declare_all_lint_rules` at the bottom.

## Rules
- [ ] feat(linter): react-perf/jsx-no-jsx-as-props
- [ ] feat(linter): react-perf/no-new-array-as-props
- [ ] feat(linter): react-perf/no-new-object-as-props
- [ ] feat(linter): react-perf/no-new-function-as-props

**Comment by Boshen** [comment]
Last task: 

- [ ] add a cli flag `--react-perf-plugin` and change all the rules to correctness.

**Comment by u1f5a4** [comment]
How do these rules for extensions work? I'm using the oxlintrc.json configuration file.

**Comment by u1f5a4** [comment]
How do these rules for Visual Studio Code extensions work? I'm using the oxlintrc.json configuration file.
